drv/fpga-devices/src/ecp5_spi_mux_pca9538.rs

```diff
@@ -254,15 +252,15 @@ impl<'a, S: SpiServer> Ecp5Driver for DeviceInstance<'a, S> {
             .config
             .configuration_port
             .lock(spi_api::CsState::Asserted)
-            .map_err(Self::Error::from)
+            .map_err(|_| Self::Error::from(SpiError::TaskRestarted))
```

Comment:
This is confusing at first glance – it's hard to know that the argument is `ServerDeath`, so it looks like it's discarding information.

What about adding `impl From<idol_runtime::ServerDeath> for SpiError` in `spi-api`, then doing `map_err(SpiError::from)` here (and everywhere else the pattern appears)?

Reply:
I was iffy about having an `impl From<idol_runtime::ServerDeath> for SpiError`, because that means that _any_ time we `?` a `ServerDeath` error, it _could_ be turned into a `SpiError` --- if we have some code that calls multiple APIs which can return `ServerDeath` errors, I didn't want them to accidentally get turned into a `SpiError` due to type inference.

Reply:
Interesting, I understand this downside but am having trouble imagining cases where that would be the wrong thing to do.

Rephrasing for my own understanding:

1) We've got a function that returns a `Result<T, SpiError>`
2) That function calls one or more Idol APIs which return `ServerDeath`
3) ...and someone puts a `?` on one of those calls
4) ...but we don't want to return `SpiError::TaskRestarted`; instead, we want a compile-time error saying that "you messed up"

In particular, needing both 3 and 4 to be true seems weird to me.

(I poked at doing `|_: idol_runtime::ServerDeath| SpiError::TaskRestarted`, but it's needlessly verbose and adds a dependency, so I think that's not great either)

Anyways, I don't feel too strongly about this!

Reply:
Oh, I was thinking more about a case where another error type implements `From<T> where T: Into<SpiError>` and also has another case that might be convertable from a `ServerDeath` error...but I'm not actually sure if such a case exists, and I think the existing blanket impls for `From` might _prevent_ us from ever being able to do that. I think I've convinced myself that it doesn't matter.

Reply:
okay, i've addressed this suggestion in c7c49f932a58a2fecb42b82cdaee23e157df311d!